lib/taskana-core/src/main/java/pro/taskana/task/internal/AttachmentHandler.java

```diff
@@ -76,18 +78,69 @@
   }
 
   void insertAndDeleteAttachmentsOnTaskUpdate(TaskImpl newTaskImpl, TaskImpl oldTaskImpl)
-      throws AttachmentPersistenceException {
-
+      throws AttachmentPersistenceException, InvalidArgumentException,
+          ClassificationNotFoundException {
     List<Attachment> newAttachments =
         newTaskImpl.getAttachments().stream().filter(Objects::nonNull).collect(Collectors.toList());
     newTaskImpl.setAttachments(newAttachments);
 
+    for (Attachment attachment : newAttachments) {
+      verifyAttachment((AttachmentImpl) attachment);
+    }
+
     deleteRemovedAttachmentsOnTaskUpdate(newTaskImpl, oldTaskImpl);
     insertNewAttachmentsOnTaskUpdate(newTaskImpl, oldTaskImpl);
     updateModifiedAttachmentsOnTaskUpdate(newTaskImpl, oldTaskImpl);
   }
 
-  void updateModifiedAttachmentsOnTaskUpdate(TaskImpl newTaskImpl, TaskImpl oldTaskImpl) {
+  void insertNewAttachmentsOnTaskCreation(TaskImpl task)
+      throws InvalidArgumentException, AttachmentPersistenceException,
+          ClassificationNotFoundException {
+    List<Attachment> attachments = task.getAttachments();
+
+    if (attachments != null) {
+      for (Attachment attachment : attachments) {
+        AttachmentImpl attachmentImpl = (AttachmentImpl) attachment;
+        verifyAttachment(attachmentImpl);
+        initAttachment(attachmentImpl, task);
+
+        try {
+          attachmentMapper.insert(attachmentImpl);
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(
+                "TaskService.updateTask() for TaskId={} INSERTED an Attachment={}.",
```

Comment:
Shouldn't this be TaskService.createTask? Also, should we maybe abstract the actual insert into a separate method, just like verifyAttachment and initAttachment?

Reply:
Yes, you're right. I'll fix this.